File: src/main/java/com/etour/app/entity/TourAddonMaster.java
package com.etour.app.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "tour_addon_master")
public class TourAddonMaster {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "tour_addon_id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "tour_id", nullable = false)
    private TourMaster tour;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "addon_id", nullable = false)
    private AddonOnMaster addon;

    @Column(name = "is_available", nullable = false)
    private Boolean isAvailable;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public TourMaster getTour() {
        return tour;
    }

    public void setTour(TourMaster tour) {
        this.tour = tour;
    }

    public AddonOnMaster getAddon() {
        return addon;
    }

    public void setAddon(AddonOnMaster addon) {
        this.addon = addon;
    }

    public Boolean getIsAvailable() {
        return isAvailable;
    }

    public void setIsAvailable(Boolean isAvailable) {
        this.isAvailable = isAvailable;
    }

}

File: dotnet/backend/Models/tour_addon_master.cs
using System;
using System.Collections.Generic;

namespace Etour_Backend_dotnet.Models;

public partial class tour_addon_master
{
    public int tour_addon_id { get; set; }

    public int tour_id { get; set; }

    public int addon_id { get; set; }

    public bool is_available { get; set; }

    public virtual addon_on_master addon { get; set; } = null!;

    public virtual tour_master tour { get; set; } = null!;
}
